src/stan/io/deserializer.hpp

```diff
@@ -0,0 +1,1978 @@
+   * between 0 and 1, incrementing the specified reference with
+   * the log of the absolute Jacobian determinant.
+   *
+   * <p>See <code>stan::math::prob_constrain(T)</code>.
+   *
+   * @tparam Ret The type to return.
+   * @tparam Jacobian Whether to increment the log of the absolute Jacobian
+   * determinant of the transform.
+   * @tparam LP Type of log probability.
+   * @tparam Sizes A parameter pack of integral types.
+   * @param lp Reference to log probability variable to increment.
+   * @param sizes Pack of integrals to use to construct the return's type.
+   * @return The next scalar transformed to a probability.
+   */
+  template <typename Ret, bool Jacobian, typename LP, typename... Sizes,
+            require_not_std_vector_t<Ret>* = nullptr>
+  auto read_prob(LP& lp, Sizes... sizes) {
+    if (Jacobian) {
+      return stan::math::prob_constrain(this->read<Ret>(sizes...), lp);
+    } else {
+      return stan::math::prob_constrain(this->read<Ret>(sizes...));
+    }
+  }
+
+  /**
+   * Specialization of `read_prob` for returning an `std::vector`.
+   *
+   * <p>See <code>stan::math::prob_constrain(T)</code>.
+   *
+   * @tparam Ret The type to return.
+   * @tparam Jacobian Whether to increment the log of the absolute Jacobian
+   * determinant of the transform.
+   * @tparam LP Type of log probability.
+   * @tparam Sizes A parameter pack of integral types.
+   * @param lp Reference to log probability variable to increment.
+   * @param sizes Pack of integrals to use to construct the return's type.
+   * @return The next scalar transformed to a probability.
+   */
+  template <typename Ret, bool Jacobian, typename LP, typename... Sizes,
+            require_std_vector_t<Ret>* = nullptr>
+  auto read_prob(LP& lp, const size_t vecsize, Sizes... sizes) {
+    std::decay_t<Ret> ret;
+    ret.reserve(vecsize);
+    for (size_t i = 0; i < vecsize; ++i) {
+      ret.emplace_back(
+          this->read_prob<value_type_t<Ret>, Jacobian>(lp, sizes...));
+    }
+    return ret;
+  }
+
+  /**
+   * Return the next object, checking that it is a valid
+   * value for a correlation, between -1 (inclusive) and
+   * 1 (inclusive).
+   *
+   * <p>See <code>stan::math::check_bounded(T)</code>.
+   *
+   * @tparam Ret The type to return.
+   * @tparam Sizes A parameter pack of integral types.
+   * @param sizes Pack of integrals to use to construct the return's type.
+   * @return Next correlation value.
+   * @throw std::runtime_error if the value is not valid
+   *   for a correlation
+   */
+  template <typename Ret, typename... Sizes,
+            require_not_matrix_t<Ret>* = nullptr,
+            require_not_std_vector_t<Ret>* = nullptr>
+  auto read_corr(Sizes... sizes) {
+    auto ret = read<Ret>(sizes...);
+    using stan::math::check_bounded;
+    using stan::math::value_of;
+    check_bounded<decltype(value_of(ret)), double, double>(
+        "deserializer", "Correlation value", value_of(ret), -1, 1);
+    return ret;
+  }
```

Comment:
I think the corr constrain only works for matrices. I don't think there is a vector corr in the language, so this can be removed.